modules/behavior/draw_way.js

```diff
@@ -423,74 +423,53 @@ export function behaviorDrawWay(context, wayID, mode, startGraph) {
-            // check if they are both part of only oneway (the same one)
-            // check if the ways that they're part of are the same way
-            // find index of the last two nodes, to determine the direction to travel around the existing way
-            // add the next node to the way we are drawing
-
             // if we're drawing an area, the first node = last node.
-            const isDrawingArea = _origWay.nodes[0] === _origWay.nodes.slice(-1)[0];
+            const isDrawingArea = _origWay.nodes.length > 1 && _origWay.nodes[0] === _origWay.nodes[_origWay.nodes.length - 1];
 
-            const [secondLastNodeId, lastNodeId] = _origWay.nodes.slice(isDrawingArea ? -3 : -2);
+            const secondLastNodeId = _origWay.nodes[_origWay.nodes.length + (isDrawingArea ? -3 : -2)];
+            const lastNodeId = _origWay.nodes[_origWay.nodes.length + (isDrawingArea ? -2 : -1)];
 
-            // Unlike startGraph, the full history graph may contain unsaved vertices to follow.
-            // https://github.com/openstreetmap/iD/issues/8749
             const historyGraph = context.history().graph();
-            if (!lastNodeId || !secondLastNodeId || !historyGraph.hasEntity(lastNodeId) || !historyGraph.hasEntity(secondLastNodeId)) {
-                context.ui().flash
-                    .duration(4000)
-                    .iconName('#iD-icon-no')
-                    .label(t.append('operations.follow.error.needs_more_initial_nodes'))();
-                return;
-            }
-
-            // If the way has looped over itself, follow some other way.
             const lastNodesParents = historyGraph.parentWays(historyGraph.entity(lastNodeId)).filter(w => w.id !== wayID);
-            const secondLastNodesParents = historyGraph.parentWays(historyGraph.entity(secondLastNodeId)).filter(w => w.id !== wayID);
-
-            const featureType = getFeatureType(lastNodesParents);
 
-            if (lastNodesParents.length !== 1 || secondLastNodesParents.length === 0) {
-                context.ui().flash
-                    .duration(4000)
-                    .iconName('#iD-icon-no')
-                    .label(t.append(`operations.follow.error.intersection_of_multiple_ways.${featureType}`))();
-                return;
+            let nextNodeIds = lastNodesParents.flatMap(({nodes}) => {
+                const indexOfLast = nodes.indexOf(lastNodeId);
+                if (indexOfLast % (nodes.length - 1)) {
+                    return [nodes[indexOfLast-1], nodes[indexOfLast+1]];
+                }
+                if (nodes[0] === nodes[nodes.length - 1]) {
+                    return [nodes[1], nodes[nodes.length - 2]];
+                }
+                return [indexOfLast ? nodes[nodes.length - 2] : nodes[1]];
+            }).filter((id) => id !== secondLastNodeId);
+
+            if (!nextNodeIds.length) return;
+
+            if (new Set([...nextNodeIds]).size > 1) {
+                if (!secondLastNodeId) {
```

Comment:
This message is currently only shown when the way doesn't have a `secondLastNode` (i.e. a stub way with only 1 node). But we should also show it if the `secondLastNode` is not part of any candidate way, like this:
```suggestion
                if (!secondLastNodeId ||
                    !lastNodesParents.some(({nodes}) => nodes.includes(secondLastNodeId))) {
```

Reply:
I'm a bit stuck and have forgotten why we even need this at all.

Can you draw up an example for
1. when `secondLastNodeId` is `false`
2. when `lastNodesParents.some(({nodes}) => nodes.includes(secondLastNodeId)` is `false`

Reply:
Tested and pushed my fix, I also moved the check after the filter that covers case 1 to cover the following two cases.

![image](https://user-images.githubusercontent.com/25524993/207973268-050a5c53-0aed-47b4-a78e-37b7d86e4d42.png)

![image](https://user-images.githubusercontent.com/25524993/207973366-7b99c337-3a07-40eb-93d4-7c07308759ec.png)